src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScanner.java

```diff
@@ -553,11 +554,49 @@ public static void visitSimpleChunks(@Nonnull Collection<FlowNode> heads, @Nonnu
         scanner.visitSimpleChunks(visitor, finder);
     }
 
-    /** Walk through flows  */
+    /** Ensures we find the last *begun* node when there are multiple heads (parallel branches)
+     *  This means that the simpleBlockVisitor gets the *actual* last node, not just the end of the last declared branch
+     *  (See issue JENKINS-38536)
+     */
+    @CheckForNull
+    private static FlowNode findLastStartedNode(@Nonnull List<FlowNode> candidates) {
+        if (candidates.size() == 0) {
+            return null;
+        } else if (candidates.size() == 1) {
+            return candidates.get(0);
+        } else {
+            FlowNode returnOut = candidates.get(0);
+            long startTime = Long.MIN_VALUE;
+            for(FlowNode f : candidates) {
+                TimingAction ta = f.getAction(TimingAction.class);
+                // Null timing with multiple heads is probably a node where the GraphListener hasn't fired to add TimingAction yet
+                long myStart = (ta == null) ? System.currentTimeMillis() : ta.getStartTime();
```

Comment:
Not an ideal presumption IMHO. One buggy node without TimingAction for a COMPLETED Pipeline will always make the node to be returned by this method. Could we just skip such nodes? 🐜